### ISSUE TITLE
训练准确率一直为0，predict结果一直为空

### ISSUE THREAD

**Issue by lonelyhwy** [chineseocr/chineseocr#498]
你好，用作者给的train data 训练ocr，还是用自己做的数据集训练ocr, loss 一直在降，但是predict的结果一直为空，准确率为0，请问是为什么呢？感谢！

**Comment by pekinghk** [comment]
我也遇到了acc=0的问题，原因应该是作者在训练加载模型的时候没有加载最后一层RNN层的权重，这当然有利有弊，如果你想要充分利用作者预训练模型的训练结果，可以把模型的所有权重都加上，这样在finetune模型的时候会有比较好的初始效果。
我改动了5个地方之后可以正常训练了，可以正常训练。
1.if  'rnn.1.embedding' not in name:##不加载最后一层权重 注释掉这一行
2.将这一行:from crnn.models.crnn import CRNN 改成: from crnn.crnn_torch import CRNN
3.如果pytorch版本比较高，将crnn/util.py中loadData函数中的v.data.resize_(data.size()).copy_(data)换成v.resize_(data.size()).copy_(data)
4.如果wrap_ctc编译不通过，可以采用高版本pytorch原生的ctc,直接将ctc的引用替换成这一句就行from torch.nn import CTCLoss，或者也可以用百度的扩展ctc。
5.把训练代码中的val函数里的pred = predict(im)这一行改成pred = predict(net, im)，同样的把这个函数def predict(im)：的输入形式改成def predict(net, im): ，然后把predict函数中的这一行：preds = model(image)改成preds = net(image)。

你可以按我说的试一试。

**Comment by fangpochen** [comment]
> 我也遇到了acc=0的问题，原因应该是作者在训练加载模型的时候没有加载最后一层RNN层的权重，这当然有利有弊，如果你想要充分利用作者预训练模型的训练结果，可以把模型的所有权重都加上，这样在finetune模型的时候会有比较好的初始效果。
> 我改动了5个地方之后可以正常训练了，可以正常训练。
> 1.if 'rnn.1.embedding' not in name:##不加载最后一层权重 注释掉这一行
> 2.将这一行:from crnn.models.crnn import CRNN 改成: from crnn.crnn_torch import CRNN
> 3.如果pytorch版本比较高，将crnn/util.py中loadData函数中的v.data.resize_(data.size()).copy_(data)换成v.resize_(data.size()).copy_(data)
> 4.如果wrap_ctc编译不通过，可以采用高版本pytorch原生的ctc,直接将ctc的引用替换成这一句就行from torch.nn import CTCLoss，或者也可以用百度的扩展ctc。
> 5.把训练代码中的val函数里的pred = predict(im)这一行改成pred = predict(net, im)，同样的把这个函数def predict(im)：的输入形式改成def predict(net, im): ，然后把predict函数中的这一行：preds = model(image)改成preds = net(image)。
> 
> 你可以按我说的试一试。

我也有这个问题。按照你的方法试了一下，为什么没有效果，acc 还是为0

**Comment by fangpochen** [comment]
> 我也遇到了acc=0的问题，原因应该是作者在训练加载模型的时候没有加载最后一层RNN层的权重，这当然有利有弊，如果你想要充分利用作者预训练模型的训练结果，可以把模型的所有权重都加上，这样在finetune模型的时候会有比较好的初始效果。
> 我改动了5个地方之后可以正常训练了，可以正常训练。
> 1.if 'rnn.1.embedding' not in name:##不加载最后一层权重 注释掉这一行
> 2.将这一行:from crnn.models.crnn import CRNN 改成: from crnn.crnn_torch import CRNN
> 3.如果pytorch版本比较高，将crnn/util.py中loadData函数中的v.data.resize_(data.size()).copy_(data)换成v.resize_(data.size()).copy_(data)
> 4.如果wrap_ctc编译不通过，可以采用高版本pytorch原生的ctc,直接将ctc的引用替换成这一句就行from torch.nn import CTCLoss，或者也可以用百度的扩展ctc。
> 5.把训练代码中的val函数里的pred = predict(im)这一行改成pred = predict(net, im)，同样的把这个函数def predict(im)：的输入形式改成def predict(net, im): ，然后把predict函数中的这一行：preds = model(image)改成preds = net(image)。
> 
> 你可以按我说的试一试。

acc 为0 的主要原因是 predict 预测都是为空
